python_modules/dagster/dagster/_core/execution/with_resources.py

```diff
@@ -55,7 +55,10 @@ def asset2(context):
                 ...
 
             asset1_with_foo, asset2_with_foo = with_resources(
-                [the_asset, other_asset],
+                [asset1, asset2],
+                resource_defs={
+                    "foo": foo_resource
```

Comment:
Normally I'd waggle a finger at `foo` being in here, but as it already seems to be everywhere in this example... 😭